src/components/CreateBook/CreateBook.jsx

```diff
@@ -67,20 +70,47 @@ const optionsGenre = [
 
 const CreateBook = ({ bookId }) => {
 
+    const routeParams = useParams();
+    console.log(routeParams)
+
     const { inputs, handleBulkInput } = useContext(InputContext);
 
     // placeholder for book update
     const [testBook, setTestBook] = useState(bookData);
 
     const [selectedImage, setSelectedImage] = useState(null);
 
-    // console.log(inputs);
-
     const handleFormSubmit = (event) => {
         event.preventDefault();
-        return (
-            console.log('this is the submit form')
-        )
+        
+        routeParams.bookId ? 
+            (updateBookAdapter( 
+                { 
+                    id: inputs.id, 
+                    title: inputs.title, 
+                    language: inputs.language, 
+                    ageRange: inputs.ageRange, 
+                    publishingYear: inputs.publishingYear, 
+                    status: inputs.status, 
+                    description: inputs.description, 
+                    genre: inputs.genre, 
+                    author: inputs.author, 
+                    worldcatURL: inputs.worldcatURL, 
+                    ISBN: inputs.ISBN, 
+                    imageURL: inputs.imageURL
+                }                   
+            )) : (createBookAdapter(
+                {
+                    title: inputs.title, 
+                    language: inputs.language, 
+                    ageRange: inputs.ageRange, 
+                    publishingYear: inputs.publishingYear, 
+                    status: inputs.status, 
+                    description: inputs.description, 
+                    genre: inputs.genre, 
+                    author: inputs.author
+                }
+            ))    
     };
 
```

Comment:
Let's add in an additional useEffect here. If routeParams.bookId is defined you should attempt to fetch the book using getSingleBookAdapter. Upon awaiting that information you should then set the returned bookData to be testBook (might want to update the name to bookInformation).

By doing this, if we are editing a book it should preload the data for it into the inputs.

Reply:
I'm getting a 401 with after implementing the suggested UseEffect. @jpremo could you please take a look?

Reply:
https://github.com/Code-the-Dream-School/bb-practicum-team1-front/pull/66/commits/6b1cf5a7d2c69d977a9e6b4430b874b64e4e4d24

Reply:
https://github.com/Code-the-Dream-School/bb-practicum-team1-front/pull/66/commits/860f7bb62716f4c2df738cc6575f8ef94ca4a71d